measures/src/main/java/io/apisense/network/ping/ICMPTask.java

```diff
@@ -110,45 +118,70 @@ private ICMPResult launchPingCommand(String command) throws PINGException, IOExc
    * @throws IOException
    */
   private ICMPResult parsePingResponse(BufferedReader stdin, long startTimeMs) throws IOException, PINGException {
-    String hostname;
-    String ip;
+    String hostname = null;
+    String ip = null;
+    Rtt rtt;
     long endTime = System.currentTimeMillis();
     long latency = endTime - startTimeMs;
     int currentTtl = config.getTtl();
 
     String line;
     Matcher matcher;
+
     while ((line = stdin.readLine()) != null) {
       Log.d(TAG, "Parsing line : " + line);
+
       matcher = PING_RESPONSE_TTL_EXCEEDED.matcher(line);
+
       if (matcher.matches()) {
         hostname = matcher.group(1);
         ip = matcher.group(2);
-        return new ICMPResult(startTimeMs, endTime, config, hostname, ip, latency, currentTtl);
+        return new ICMPResult(startTimeMs, endTime, config, hostname, ip, latency, currentTtl, null);
       }
+
       matcher = PING_RESPONSE_TTL_EXCEEDED_NO_HOSTNAME.matcher(line);
+
       if (matcher.matches()) {
         ip = matcher.group(1);
-        return new ICMPResult(startTimeMs, endTime, config, null, ip, latency, currentTtl);
+        return new ICMPResult(startTimeMs, endTime, config, null, ip, latency, currentTtl, null);
       }
+
       matcher = PING_RESPONSE_SUCCESS.matcher(line);
+
       if (matcher.matches()) {
         hostname = matcher.group(1);
         ip = matcher.group(2);
         latency = Float.valueOf(matcher.group(3)).longValue(); // milliseconds
-        return new ICMPResult(startTimeMs, endTime, config, hostname, ip, latency, currentTtl);
       }
+
       matcher = PING_RESPONSE_SUCCESS_NO_HOSTNAME.matcher(line);
+
       if (matcher.matches()) {
         ip = matcher.group(1);
         latency = Float.valueOf(matcher.group(2)).longValue(); // milliseconds
-        return new ICMPResult(startTimeMs, endTime, config, null, ip, latency, currentTtl);
+        hostname = null;
```

Comment:
This match is exclusive with the previous one, so it's useless to set hostname back to null.